From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 22 Nov 2022 20:05:57 +0100
Subject: [PATCH] fix building with QT_FEATURE_qml_devtools disabled

Without this, building fails with:

CMake Error at qtbase/cmake/QtPluginHelpers.cmake:496 (message):
  The plug-in 'QmlLintQuickPlugin' does not belong to any Qt module.
Call Stack (most recent call first):
  qtbase/cmake/QtPluginHelpers.cmake:189 (qt_internal_get_module_for_plugin)
  qtdeclarative/src/plugins/qmllint/quick/CMakeLists.txt:1 (qt_internal_add_plugin)

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 qtdeclarative/src/plugins/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/qtdeclarative/src/plugins/CMakeLists.txt b/qtdeclarative/src/plugins/CMakeLists.txt
index 548ef486f014..54058d431934 100644
--- a/qtdeclarative/src/plugins/CMakeLists.txt
+++ b/qtdeclarative/src/plugins/CMakeLists.txt
@@ -10,4 +10,6 @@ if(TARGET Qt::Quick)
     add_subdirectory(scenegraph)
 endif()
 
-add_subdirectory(qmllint)
+if(QT_FEATURE_qml_devtools)
+    add_subdirectory(qmllint)
+endif()
